doc/examples/segmentation/plot_rolling_ball.py

```diff
@@ -0,0 +1,150 @@
+        working_img = 255 - working_img
+
+    # tensor representation of the image
+    # (stacked blocks as described in the paper)
+    background = (np.arange(255)[np.newaxis,
+                                 np.newaxis, :] < working_img[..., np.newaxis])
+
+    # roll the ball around
+    # here: implemented as a sequence of small balls (B1, ..., Bk)
+    # because the current implementation of a closure with a large ball
+    # is slow and consumes a lot of memory
+    selem = ball(1)
+    for idx in range(radius):
+        background = erosion(background, selem)
+
+    for idx in range(radius):
+        background = dilation(background, selem)
+
+    # flatten the tensor
+    background = np.argmax(background == False, axis=-1)
+
+    filtered_image = working_img - background
+
+    if white_background:
+        filtered_image = 255 - filtered_image
+        background = 255 - background
+
+    return filtered_image, background
+
+
+######################################################################
+# And then test it using an image with white background
+
+import skimage.data as data
+import matplotlib.pyplot as plt
+
+
+image = data.page()
+
+filtered_image, background = rolling_ball(image, radius=30,
+                                          white_background=True)
+fig, ax = plt.subplots(nrows=1, ncols=3)
+
+ax[0].imshow(image, cmap='gray')
+ax[0].set_title('Original image')
+
+ax[1].imshow(background, cmap='gray')
+ax[1].set_title('Background')
+
+ax[2].imshow(filtered_image, cmap='gray')
+ax[2].set_title('Result')
+
+plt.show()
+
+######################################################################
+# And with an image with black background
+
+image = data.coins()
+
+filtered_image, background = rolling_ball(image, radius=30,
+                                          white_background=True)
+fig, ax = plt.subplots(nrows=1, ncols=3)
+
+ax[0].imshow(image, cmap='gray')
+ax[0].set_title('Original image')
+
+ax[1].imshow(background, cmap='gray')
+ax[1].set_title('Background')
+
+ax[2].imshow(filtered_image, cmap='gray')
+ax[2].set_title('Result')
+
+plt.show()
+
+######################################################################
+# As you may have noticed, above implementation is rather slow.
+# If runtime is a concern, we can implement an efficient approximation
+# using a tophat filter and a disk element
+
+from skimage.morphology import disk, opening, black_tophat, white_tophat
+
+
+def rolling_ball(image, radius=50, white_background=False):
+    selem = disk(radius)
+
+    background = opening(image, selem)
+
+    if white_background:
+        filtered_image = black_tophat(image, selem)
+    else:
+        filtered_image = white_tophat(image, selem)
+
+    return filtered_image, background
+
+######################################################################
+# Which produces almost identical results to the original algorithm
+# albeit being much faster.
+
+import skimage.data as data
+import matplotlib.pyplot as plt
```

Comment:
Above imports are already in.

Reply:
Yes, my line of thinking here is that each code snippet can be directly copy-pasted into, say, a jupyter notebook and will have the necessary imports with it. If this is not the style of the repo, we can ofc change it.

Reply:
I figured, actually, and I like it in absolute terms. I think the style of the gallery examples is more linear (as in *narrative* docs). Please @scikit-image/core correct me if I'm off. Also, readers can readily download the example as a Jupyter notebook -- now maybe you're thinking "but still" ;)

Reply:
+ or launch Binder to run it without downloading it

Reply:
It looks like the GitHub handle I tried didn't parse, although I'm pretty sure I copied-pasted it from a comment by @jni somewhere... Trying otherwise, as seen in a comment by @emmanuelle : @skimage/core No, the tag is still not working! :thinking:

Reply:
@mkcor only core devs can summon the core devs. This is a GH design choice I disagree with, though I understand why they did it. 😞 

I agree with @mkcor, I think the document should make sense top to bottom, not necessarily as individual snippets. I'd remove the "redundant" imports. Typically notebooks are not expected to have each cell run independently of the others — and these documents have the same idea.

Reply:
> only core devs can summon the core devs. This is a GH design choice I disagree with, though I understand why they did it. :disappointed:

Oh, good to know! Thanks for clarifying @jni :)

Reply:
fixed - I reordered the imports and moved them all to the top to better comply with PEP8